### ISSUE TITLE
Auto-focus search box in Add Tag window when external keyboard is attached?

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero-ios#688]
https://forums.zotero.org/discussion/105027/enhancement-for-ios-auto-focus-on-the-tag-search-field-when-searching-tags

I haven't tested with an external keyboard, but if this wouldn't interfere with tapping, this would make it easier to just start typing.

**Comment by mvasilak** [comment]
I tested with the MacBook keyboard connected to an iPad, and I cannot undestand what is the exact issue. 
I very easily can press `tab` to switch focus to visible textfields, therefore tag search is 1 or 2 keystrokes away,
since by default it's visible in the tags pane.

Does the user mean something else, e.g. pressing the search button in the external keyboard, and the default behaviour (showing the spotlight search in the iPad) is overriden to focus on an app search field instead? Could we ask them for some more info, ideally a video presenting the issue?

**Comment by dstillman** [comment]
They might just not know you can press Tab, and I imagine many people wouldn't think of that. Regardless, I think the request would still be for the pane to start with focus in the search field when a keyboard is attached. Maybe there's an argument for consistency — we wouldn't focus the text field for all users — but it seems reasonable to prioritize a text field in some situations when a keyboard is attached.

**Comment by mvasilak** [comment]
Since the `Search Items` field is also visible by default, perhaps it's this one that should gain focus. Ideally we could track the last search field used, and give focus to that one.

**Comment by dstillman** [comment]
No, this is just for adding tags. We wouldn't do this for the items list.

**Comment by mvasilak** [comment]
Did some research, and there are some solutions with different pros and cons:
* Simply make the search field the first responder, either every time the view did appear, or just for the first time. This will work independently of the type of keyboard, and may not be ideal for those that just don't need to search.
* Use https://developer.apple.com/documentation/gamecontroller/gckeyboard to get notifications when an external keyboard did connect/disconnect. However, w/ no immediate access to an actual external keyboard for iPad, it cannot be tested precisely. Simulator "keyboard dis/connection" doesn't always work as expected...
* Use https://developer.apple.com/documentation/uikit/uigesturerecognizer/1619995-pressesbegan to detect when a physical button has been pressed, and make then first responder the search field. The caveat is that while it works ideally, unless the user tries it they won't know it is possible, because the focus won't be in the search field.

I'll try to get an external keyboard to properly test. Any feedback until then is welcome.
